File: editor/en/[2958]Length of Longest Subarray With at Most K Frequency.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int maxSubarrayLength(int[] nums, int k) {
        Map<Integer, Integer> map = new HashMap<>();
        int max = 0;
        int left = 0;

        for (int right = 0; right < nums.length; right++) {
            map.put(nums[right], map.getOrDefault(nums[right], 0) + 1);

            while (map.get(nums[right]) > k) {
                map.put(nums[left], map.get(nums[left]) - 1);
                left++;
            }

            max = Math.max(max, right - left + 1);
        }

        return max;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[2958]Length of Longest Subarray With at Most K Frequency.py
from collections import defaultdict


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def maxSubarrayLength(self, nums: List[int], k: int) -> int:
        freq = defaultdict(int)
        max_length = 0
        left = 0

        for right in range(len(nums)):
            freq[nums[right]] += 1
            while freq[nums[right]] > k:
                freq[nums[left]] -= 1
                left += 1
            max_length = max(max_length, right - left + 1)

        return max_length

# leetcode submit region end(Prohibit modification and deletion)
